test/production/app-dir-edge-runtime-with-wasm/index.test.ts

```diff
@@ -0,0 +1,62 @@
+import path from 'path'
+import { createNext, FileRef } from 'e2e-utils'
+import { NextInstance } from 'test/lib/next-modes/base'
+import { renderViaHTTP } from 'next-test-utils'
+
+const files = {
+  'app/layout.tsx': `
+    export default function AppLayout({ children }) {
+      return (
+        <html>
+          <head>
+            <title>WASM Import</title>
+          </head>
+          <body>
+            {children}
+          </body>
+        </html>
+      )
+    }
+  `,
+  'app/page.tsx': `
+    // @ts-expect-error
+    import wasm from '../wasm/add.wasm?module'
+
+    console.log(wasm)
+
+    export default function Page() {
+      return "index page"
+    }
+
+    export const runtime = "experimental-edge"
+  `,
+  'wasm/add.wasm': new FileRef(path.join(__dirname, 'add.wasm')),
+}
+
+describe('app-dir edge runtime with wasm', () => {
```

Comment:
I'm not sure if it would be preferred to have all of the `app-dir` related tests in `test/e2e/app-dir`, let me know!

Reply:
I think it's good. We can always merge tests. Extracting is harder. :pray: